mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulProfileScreen.kt

```diff
@@ -53,7 +55,10 @@ class StatefulProfileScreen(
     }
 
     scope.launch {
-      puzzles = chessFacade.solvedPuzzles(user).map { it.toPuzzleInfoAdapter() }.sortedBy { it.elo }
+      chessFacade
+          .solvedPuzzles(user)
+          .onEach { flow -> puzzles = flow.map { it.toPuzzleInfoAdapter() }.sortedBy { it.elo } }
```

Comment:
I think you're handling a `List` here, and not a `Flow`

```suggestion
          .onEach { list -> puzzles = list { it.toPuzzleInfoAdapter() }.sortedBy { it.elo } }
```

Reply:
https://github.com/epfl-SDP/android/pull/450/commits/49543540a19b86a0b1a17c1e91e3217b4a346e72